Source/RMG/UserInterface/Widget/RomBrowserWidget.cpp

```diff
@@ -460,15 +448,27 @@ void RomBrowserWidget::customContextMenuRequested(QPoint position)
 
     this->getCurrentData(data);
 
+    this->mainContextMenu->clear();
+
+    this->mainContextMenu->addAction(this->action_PlayGame);
+    this->mainContextMenu->addAction(this->action_PlayGameWith);
+    this->mainContextMenu->addSeparator();
+    this->mainContextMenu->addAction(this->action_RefreshRomList);
+    this->mainContextMenu->addSeparator();
+    this->mainContextMenu->addAction(this->action_OpenRomDirectory);
+    this->mainContextMenu->addAction(this->action_ChangeRomDirectory);
+    this->mainContextMenu->addSeparator();
+    this->mainContextMenu->addAction(this->action_RomInformation);
+    this->mainContextMenu->addSeparator();
+    this->mainContextMenu->addAction(this->action_EditGameSettings);
+    this->mainContextMenu->addAction(this->action_EditCheats);
+    this->mainContextMenu->addSeparator();
+
     this->action_PlayGame->setEnabled(hasSelection);
     this->action_PlayGameWith->setEnabled(hasSelection);
     this->action_RomInformation->setEnabled(hasSelection);
     this->action_EditGameSettings->setEnabled(hasSelection);
     this->action_EditCheats->setEnabled(hasSelection);
-    this->action_SetCoverImage->setEnabled(hasSelection);
-    this->action_SetCoverImage->setVisible(view == this->gridViewWidget);
-    this->action_RemoveCoverImage->setEnabled(hasSelection && !data.coverFile.isEmpty());
-    this->action_RemoveCoverImage->setVisible(view == this->gridViewWidget);
```

Comment:
Please also revert this change and add the new columns qmenu below this in the same way as it was done here before your removal.

Reply:
Done, will include in next commit. (I implemented the submenu this way because I was having trouble setting its visibility. I was trying to call "setVisible()" on the submenu itself when I should have been calling it on the submenu's "menuAction()". Now that I've figured that out, I've implemented the submenu the same way as all of the other actions in the context menu.)